event-consumer/actions/generic/index.js

```diff
@@ -0,0 +1,148 @@
+/*
+* <license header>
+*/
+
+/**
+ * This is a sample action showcasing how to access an external API
+ *
+ * Note:
+ * You might want to disable authentication and authorization checks against Adobe Identity Management System for a generic action. In that case:
+ *   - Remove the require-adobe-auth annotation for this action in the manifest.yml of your application
+ *   - Remove the Authorization header from the array passed in checkMissingRequestInputs
+ *   - The two steps above imply that every client knowing the URL to this deployed action will be able to invoke it without any authentication and authorization checks against Adobe Identity Management System
+ *   - Make sure to validate these changes against your security requirements before deploying the action
+ */
+
+const fetch = require('node-fetch')
+const { Core, Events } = require('@adobe/aio-sdk')
+const { context, getToken } = require('@adobe/aio-lib-ims')
+const stateLib = require('@adobe/aio-lib-state')
+const { errorResponse, getBearerToken, stringParameters, checkMissingRequestInputs } = require('../utils')
+
+const slackWebhook = "https://hooks.slack.com/services/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"
+const slackChannel = "Jie Yu"
+const request = require('request')
+
+
+
+async function sendToSlack(msg) {
+  var slackMessage = " Event received: " + msg;
+  return new Promise(function (resolve, reject) {
+    var payload = {
+      "channel": slackChannel,
+      "username": "incoming-webhook",
+      "text": slackMessage,
+      "mrkdwn": true,
+    }
+
+    var options = {
+      method: 'POST',
+      url: slackWebhook,
+      headers:
+          { 'Content-type': 'application/json' },
+      body: JSON.stringify(payload)
+    }
+
+    request(options, function (error, response, body) {
+      if (error) {
+        reject(error)
+
+      } else {
+        resolve(response)
+      }
+
+    })
+  })
+}
+
+async function fetchEvent(params, token, since) {
+  const eventsClient = await Events.init(params.ims_org_id, params.apiKey, token)
+
+  if (since == null) {
+    journalling = await eventsClient.getEventsFromJournal(params.journalling_url)
+  } else {
+    journalling = await eventsClient.getEventsFromJournal(params.journalling_url, {since: since})
+  }
+  
+  return journalling.events[0]
```

Comment:
We can return the entire batch of events from here instead of returning the first event only. wdyt?
I am assuming we are not specifying limit=1 in the Journaling API (https://www.adobe.io/apis/experienceplatform/events/docs.html#!adobedocs/adobeio-events/master/api/journaling_api.md#limiting-the-size-of-the-batch).

Reply:
in my testing, I never get the result more than one, even there are more several events in the event queue. but i will return events instead of events[0] here

Reply:
The number of events in a batch depends on the incoming load. Thanks for supporting multiple-events in a go. 👍